openedx/core/djangoapps/credentials_service/serializers.py

```diff
@@ -0,0 +1,84 @@
+"""
+Serializers for data manipulated by the credentials service APIs.
+"""
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied
+from openedx.core.djangoapps.credentials_service.models import UserCredential, AbstractCredential, \
+    UserCredentialAttribute, ProgramCertificate, CourseCertificate
+
+
+class CredentialRelatedField(serializers.RelatedField):
+    """
+    A custom field to use for the user credential generic relationship.
+    """
+    def to_representation(self, value):
+        """
+        Serialize objects to a simple textual representation.
+        """
+        if isinstance(value, ProgramCertificate):
+            return value.program_id
+        elif isinstance(value, CourseCertificate):
+            return {
+                'course_id': value.course_id,
+                'certificate_type': value.certificate_type
+            }
```

Comment:
is it intentional that the program relation only returns a scalar field while the course relation returns a nested object?  Having the program relation return the id in a nested object will be easier to extend with additional attributes in the future (if useful), and will simplify parsers if they need to handle both types.